File: Coding/TS/Day1P2_SW.ts
const s = "RRBBRRBRBRRBRRBRRB";
const L = 6;
let res = L;

function getAnswer(s: string, L:number){
    let cnt = 0;
    for(let i=0;i<L;i++){
        if(s[i]==='B') cnt++;
    }
    res = Math.min(res, L-cnt);
    for(let i=L;i<s.length;i++){
        if(s[i-L]==='B') cnt--;
        if(s[i]==='B') cnt++;
        res = Math.min(res, L-cnt);
    }
    return res;
}

console.log(getAnswer(s, L));

File: Coding/java/Day1P2_SW.java
package Coding.java;
import java.util.*;
public class Day1P2_SW {
    public static void main(String[] args){
        Scanner sc = new Scanner(System.in);
        String s = sc.next();
        int L = sc.nextInt();
        int res = L;
        int cnt = 0;
        for(int i=0;i<L;i++){
            if(s.charAt(i)=='B') cnt++;
        }
        res = Math.min(res, L-cnt);
        for(int i=L;i<s.length();i++){
            if(s.charAt(i-L)=='B') cnt--;
            if(s.charAt(i)=='B') cnt++;
            res = Math.min(res, L-cnt);
        }
        System.out.println(res);
        sc.close();
    }
}